nixos/doc/manual/development/reviewing-contributions.xml

```diff
@@ -0,0 +1,257 @@
+
+<itemizedlist>
+  <listitem><para>Add labels to the pull-request.</para>
+    <itemizedlist>
+      <listitem><para><literal>8.has: module (new)</literal> and any topic label that fit the module.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Checking that the module tests, if any, are succeeding.</para></listitem>
+  <listitem><para>Checking that the introduced options are correct.</para>
+    <itemizedlist>
+      <listitem><para>Type should be appropriate (string related types differs in their merging capabilities, <literal>optionSet</literal> and <literal>string</literal> types are deprecated).</para></listitem>
+      <listitem><para>Description, default and example should be provided.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Checking that module <literal>meta</literal> field is present</para>
+    <itemizedlist>
+      <listitem><para>Maintainers should be declared in <literal>meta.maintainers</literal>.</para></listitem>
+      <listitem><para>Module documentation should be declared with <literal>meta.doc</literal>.</para></listitem>
+    </itemizedlist>
+  </listitem>
+  <listitem><para>Checking that the module respect other modules functionality.</para>
+    <itemizedlist>
+      <listitem><para>For example, enabling a module should not open firewall ports by default.</para></listitem>
+    </itemizedlist>
+  </listitem>
+</itemizedlist>
+
+<example><title>Sample template for a new module review</title>
+<screen>
+##### Reviewed points
+
+- [ ] module path fits the guidelines
+- [ ] module tests succeed on ARCHITECTURE
+- [ ] options have appropriate types
+- [ ] options have default
+- [ ] options have example
+- [ ] options have descriptions
+- [ ] No unneeded package is added to system.environmentPackages
+- [ ] meta.maintainers is set
+- [ ] module documentation is declared in meta.doc
+
+##### Possible improvements
+
+##### Comments
+
+</screen></example>
+</section>
+
+<section><title>Other submissions</title>
+
+<para>Other type of submissions requires different reviewing steps.</para>
+
+<para>If you consider having enough knowledge and experience in a topic and would like to be a long-term reviewer for related submissions, please contact the current reviewers for that topic. They will give you information about the reviewing process.
+The main reviewers for a topic can be hard to find as there is no list, but checking past pull-requests to see who reviewed or git-blaming the code to see who committed to that topic can give some hints.</para>
+
+<para>Container system, boot system and library changes are some examples of the pull requests fitting this category.</para>
+
+</section>
+
+<section><title>Merging pull-requests</title>
+
+<para>It is possible for community members that have enough knowledge and experience on a special topic to contribute by merging pull requests.</para>
+
+<para>TODO: add the procedure to request merging rights.</para>
```

Comment:
I would definitely like to know this. I was added to the NixOS org mostly for labeling Issues and PRs but would be interested in merging some PRs but want to know the guidelines and get permission.